addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -448,25 +450,51 @@ export const editorCommands = {
     },
     unlink: editor => {
         const sel = editor.document.getSelection();
-        // we need to remove the contentEditable isolation of links
-        // before we apply the unlink, otherwise the command is not performed
-        // because the content editable root is the link
-        const closestEl = closestElement(sel.focusNode, 'a');
-        if(closestEl) {
-            closestEl.removeAttribute('class'); // To prevent firefox from adding unnecessary <span>.
-        }
-        if (closestEl && closestEl.getAttribute('contenteditable') === 'true') {
-            editor._activateContenteditable();
+        const isCollapsed = !!sel.isCollapsed;
+        if (!isCollapsed) {
+            getDeepRange(editor.editable, { sel, splitText: true, select: true });
+            // Split leading and trailing links.
+            let { anchorNode, focusNode, anchorOffset, focusOffset } = sel;
+            const direction = getCursorDirection(anchorNode, anchorOffset, focusNode, focusOffset);
+            let startLink = closestElement(anchorNode, 'a');
+            if (startLink) {
+                startLink = splitAroundUntil(anchorNode, startLink);
+                const range = sel.getRangeAt(0);
+                anchorNode = startLink;
+                anchorOffset = direction === DIRECTIONS.RIGHT ? 0 : nodeSize(startLink);
+                range.setStart(anchorNode, anchorOffset);
+                sel.removeAllRanges();
+                sel.addRange(range);
+                sel.extend(focusNode, focusOffset);
+                getDeepRange(editor.editable, { sel, select: true });
```

Comment:
I'm confused... this is a lot of lines for what exactly ? Calling `splitAroundUtil` ? Then what ? The selection is log somehow so we need to reset it or something ? If that's the case can't we just clone the selection data and restore it later using the `setSelection` util ?

Reply:
This is all to handle cases like [this](https://github.com/odoo/odoo/pull/85339/commits/b4297849b8e8c6b22bc7c4611caa17eec6528f2f#diff-9ecae2edd5662a0c41ae89c5d0d726c5511d4d90d096648f62d6855941f148d4R447) (pre-existing) test:

```js
await testEditor(BasicEditor, {
    contentBefore: '<p>a<a href="exist">b[c]d</a>e</p>',
    stepFunction: async editor => {
        await unlink(editor);
    },
    contentAfter: '<p>a<a href="exist">b[</a>c]<a href="exist">d</a>e</p>',
});
```

We need to split the link(s) so we only unlink what's selected, and then keep the selection.

Reply:
Ok thanks, I understand the need for `splitAroundUtil` and such. Can't we use the `setSelection` util though ?